tfx/experimental/pipeline_testing/pipeline_recorder_utils_test.py

```diff
@@ -0,0 +1,144 @@
+# Lint as: python3
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for tfx.experimental.pipeline_testing.pipeline_recorder_utils."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import mock
+import tensorflow as tf
+
+from tfx.experimental.pipeline_testing import pipeline_recorder_utils
+from tfx.utils import io_utils
+
+class PipelineRecorderUtilsTest(tf.test.TestCase):
+
+  def setUp(self):
+    super(PipelineRecorderUtilsTest, self).setUp()
+    self._base_dir = os.path.join(self.get_temp_dir(), 'base_dir')
+    self.src_uri = os.path.join(self._base_dir, 'input')
+    self.dest_uri = os.path.join(self._base_dir, 'output')
+    tf.io.gfile.makedirs(self.src_uri)
+    tf.io.gfile.makedirs(self.dest_uri)
+
+    # Writing a string to test.txt file in src_uri
+    self.content = "pipeline recorded"
+    io_utils.write_string_file(os.path.join(self.src_uri, 'test.txt'),
+                               self.content)
+
+    # Placeholders for record_pipeline(...) arguments
```

Comment:
It looks like that current test really use metadata connection in https://github.com/tensorflow/tfx/blob/master/tfx/orchestration/metadata.py#L147. Can we replace metadata with mock object?  More specifically, this test fails if we cannot create a file called "metadata_db_uri" which is true for google internal environment.